### PR TITLE
[ValloxMV] Switched to shared instance of WebSocketClient

### PR DESCRIPTION
**Pull request by bjoernbrings**
Switched to shared instance of WebSocketClient using WebSocketFactory

Tried to follow example #4164 from @cweitkamp as much as possible.

Signed-off-by: Björn Brings <bjoernbrings@web.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java [hunk 1]
```diff
@@ -29,10 +31,9 @@
  * @author Björn Brings - Initial contribution
  */
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.valloxmv")
-@NonNullByDefault
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java]: webSocketClient could become null. So I looked into the Kodi code, where there was no NonNullByDefault Anotation.
Could go for another solution: Left the generell anotation and just applied it for this one variable.

File: addons/binding/org.openhab.binding.valloxmv/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -33,3 +33,4 @@ Import-Package:
  org.osgi.service.component.annotations;resolution:=optional,
  org.slf4j
 Service-Component: OSGI-INF/*.xml
+Require-Bundle: org.eclipse.smarthome.io.net
```
[the remark this answers is not in the document]
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/META-INF/MANIFEST.MF]: Hmm, that was done by Eclipse automatically. Changed it as suggested

**Comment by bjoernbrings** [comment]
Thanks for your quick feedback. Hope new commit resolves it.

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java [hunk 3]
```diff
@@ -31,8 +34,9 @@
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.valloxmv")
 @NonNullByDefault
 public class ValloxMVHandlerFactory extends BaseThingHandlerFactory {
+    @Nullable
+    private WebSocketClient webSocketClient;
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java]: Sry to be nit-picking but our guidelines tell us to place the annotation between the keyword `private` and `WebSocketClient`.

Maybe the better choice in this case is `@NonNullByDefault({})`.

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java [hunk 4]
```diff
@@ -51,8 +53,9 @@
     /**
      * IP of vallox ventilation unit web interface.
      */
-    public ValloxMVHandler(@NonNull Thing thing) {
+    public ValloxMVHandler(@NonNullByDefault Thing thing, WebSocketClient webSocketClient) {
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java]: This is not exactly what I mean. What I wanted to express with the term "on class level" is that the annotation should be added in front of the `class` keyword like in the `ValloxMVHandlerFactory`.

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java (hunk 3 above)
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandlerFactory.java]: As I get for the second option an warning from eclipse which I was neither able to handle ("Nullness default is redundant with a default specified for the enclosing package org.openhab.binding.valloxmv.internal") nor I had an idea of the result, I implemented the first suggestion.

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java (hunk 4 above)
Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java]: Makes sense. As reasult I had to do some (maybe usefull) null checks as well.

File: addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java [hunk 5]
```diff
@@ -36,11 +38,13 @@
...
  */
+@NonNullByDefault
 public class ValloxMVHandler extends BaseThingHandler {
 
     private final Logger logger = LoggerFactory.getLogger(ValloxMVHandler.class);
-    private ScheduledFuture<?> updateTasks;
-    private ValloxMVWebSocket valloxSendSocket;
+    private @Nullable ScheduledFuture<?> updateTasks;
+    private @Nullable ValloxMVWebSocket valloxSendSocket;
+    private @Nullable WebSocketClient webSocketClient;
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java]: One final comment: I consider this not to be `@Nullable`.

Comment (bjoernbrings) [reply addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java]: You're right. Changed it.

**Pull request merged**